src/Application/Digital/Commands/ShareEntry.cs

```diff
@@ -0,0 +1,178 @@
+using System.Configuration;
+using Application.Common.Exceptions;
+using Application.Common.Interfaces;
+using Application.Common.Models.Dtos.Digital;
+using Application.Common.Models.Operations;
+using AutoMapper;
+using Domain.Entities;
+using Domain.Entities.Digital;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using NodaTime;
+
+namespace Application.Digital.Commands;
+
+public class ShareEntry
+{
+    public record Command : IRequest<EntryPermissionDto>
+    {
+        public User CurrentUser { get; init; } = null!;
+        public Guid EntryId { get; init; }
+        public Guid UserId { get; init; }
+        public DateTime ExpiryDate { get; init; }
+        public bool CanView { get; init; }
+        public bool CanUpload { get; init; }
+        public bool CanDownload { get; init; }
+        public bool CanChangePermission { get; init; }
+    }
+    
+    public class CommandHandler : IRequestHandler<Command, EntryPermissionDto>
+    {
+        private readonly IApplicationDbContext _context;
+        private readonly IMapper _mapper;
+        private readonly IDateTimeProvider _dateTimeProvider;
+
+        public CommandHandler(IApplicationDbContext applicationDbContext, IMapper mapper, IDateTimeProvider dateTimeProvider)
+        {
+            _context = applicationDbContext;
+            _mapper = mapper;
+            _dateTimeProvider = dateTimeProvider;
+        }
+        
+        public async Task<EntryPermissionDto> Handle(Command request, CancellationToken cancellationToken)
+        {
+            var entry = await _context.Entries
+                .Include(x => x.Uploader)
+                .Include(x => x.Owner)
+                .FirstOrDefaultAsync(x => x.Id == request.EntryId, cancellationToken);
+            
+            if (entry is null)
+            {
+                throw new KeyNotFoundException("Entry does not exist.");
+            }
+
+            var hasModifyPermission = _context.EntryPermissions.Any(x =>
```

Comment:
I don't think the naming quite captures the intention of the code

Reply:
Any suggestion?

Reply:
How about `cuerrentUserCanChangePermissionOfEntry`

Reply:
`canChangePermission` would be sufficient

Reply:
Fixed at https://github.com/ProFile-org/ProFile-BE/pull/336/commits/7fe548099a1708b6ff07e2e7cc7b0b7677a8af23